src/progress/inflights.rs

```diff
@@ -97,27 +97,34 @@ impl Inflights {
             // out of the left side of the window
             return;
         }
-
-        let mut i = 0usize;
-        let mut idx = self.start;
-        while i < self.count {
-            if to < self.buffer[idx] {
-                // found the first large inflight
+        let mut start = self.start;
+        let mut end = self.start + self.count - 1;
```

Comment:
It's wrong to compute the end for a ring queue.

Reply:
Oh, that's right. I'll fix this and add some test cases for this.

Reply:
Here I just treat it like a normal array since we have `start` and `count`. And all the indexing things are wrapped by `self.rotate`. So I think we have safety here? @BusyJay

Reply:
Indeed.